client/src/components/Chat.jsx

```diff
@@ -255,16 +282,16 @@ const Chat = () => {
 	}, [editing]);
 
 	useEffect(() => {
-		const newMessageHandler = (message) => {
+		const newMessageHandler = async (message) => {
 			try {
+				const decryptedMessage = await decryptMessage(message.message, cryptoKey);
 				addMessage(message);
 				playNotification('newMessage');
-				createBrowserNotification('You received a new message on Whisper', message.message);
+				createBrowserNotification('You received a new message on Whisper', decryptedMessage);
```

Comment:
This is a problem you either decrypt it here and remove the use of decrypt in the function itself, or decrypt it in the function(preferably this), like originally done. You should thoroughly test all places where a message is used to see if you tackled all issues.
https://github.com/Dun-sin/Whisper/blob/15ba8c38aee3015ffb4ac824dd06a2a591065920/client/src/lib/browserNotification.js#L24-L34

Reply:
sorry didnt saw that browser notifications also had bad words check inside will fix it now.

Reply:
no problem, that's why we have code reviews

Reply:
i was gonna decrypt it inside function but  we need cryptoKey for it and passing it into function like createBrowserNotification felt inapporpriate so removed the decrypt from the createBrowserNotifications. And just wanted to say thing that badwords.filter function isnt working on my side means i tried putting some random strings too but it gave undefined. Not related to my issue but just wanted you to know.

Reply:
Alright